gapic_smoketest.py

```diff
@@ -0,0 +1,178 @@
+    # Run gradle test in the main java directory and also in the gapic directory.
+    cmds = ("%s/gradlew build test" % os.getcwd()).split()
+    exit_code = subprocess.call(cmds, stdout=log, stderr=log)
+    if exit_code:
+        return exit_code
+
+    cmds = ("%s/gradlew -p %s build test" % (os.getcwd(), gapic_dir)).split()
+    return subprocess.call(cmds, stdout=log, stderr=log)
+
+def run_smoke_test(api_name, language, root_dir, log, user_config):
+    log_file = _setup_logger(log)
+    failure = []
+    success = []
+    warning = []
+    (api_version, artman_yaml_path) = test_apis[api_name]
+    target = language + "_gapic"
+
+    # Generate client library for an API and language.
+    if _generate_artifact(artman_yaml_path,
+                          target,
+                          root_dir,
+                          log_file,
+                          user_config):
+        msg = 'Failed to generate %s of %s.' % (
+            target, artman_yaml_path)
+        failure.append(msg)
+        logger.info(msg)
+    else:
+        msg = 'Succeded to generate %s of %s.' % (
+            target, artman_yaml_path)
+        success.append(msg)
+        logger.info(msg)
+
+        # Test the generated client library.
+        logger.info("Starting testing of %s %s client library." %(language, api_name))
+        cwd = os.getcwd()
+        os.chdir("artman-genfiles/%s" % language)
+
+        if _test_artifact(test_languages[language], api_name, api_version, log_file):
+            msg = 'Failed to pass tests for %s library.' % (
+                language)
+            failure.append(msg)
+        else:
+            msg = 'Succeeded to pass tests for %s library.' % (
+                language)
+            success.append(msg)
+        os.chdir(cwd)
+        logger.info(msg)
+    logger.info('================ Smoketest summary ================')
+    logger.info('Successes:')
+    for msg in success:
+        logger.info(msg)
+    logger.info('Warnings:')
+    if warning:
+        for msg in warning:
+            logger.info(msg)
+    else:
+        logger.info("none.")
+    logger.info('Failures:')
+    if failure:
+        for msg in failure:
+            logger.error(msg)
+        sys.exit('Smoke test failed.')
+    else:
+        logger.info("none.")
+
+
+def _generate_artifact(artman_config, artifact_name, root_dir, log_file, user_config_file):
+    with open(log_file, 'a') as log:
+        grpc_pipeline_args = [
+            'artman',
+            '--local',
+            '--verbose',
+        ]
+        if user_config_file:
+            grpc_pipeline_args = grpc_pipeline_args + [
+                '--user-config', user_config_file,]
+        grpc_pipeline_args = grpc_pipeline_args + [
+            '--config', os.path.join(root_dir, artman_config),
+            '--root-dir', root_dir,
+            'generate', artifact_name
+        ]
+        logger.info("Start artifact generation for %s of %s: %s" %
```

Comment:
nit: s/Start artifact generation for/Generate artifact

Reply:
Done.